src/index.js

```diff
@@ -5,29 +5,35 @@ import {
   Routes,
   Route,
 } from 'react-router-dom';
+import { Provider } from 'react-redux';
 import './index.css';
+import store from './redux/configureStore';
 import App from './App';
 import Books from './components/Books';
 import Categories from './components/Categories';
 import reportWebVitals from './reportWebVitals';
 
 ReactDOM.render(
   <React.StrictMode>
-    <Router>
-      <Routes>
-        <Route path="/" element={<App />}>
-          <Route index element={<Books />} />
-          <Route path="books" element={<Books />} />
-          <Route path="categories" element={<Categories />} />
-        </Route>
-        <Route
-          path="*"
-          element={
-            <h1>Nothing here!</h1>
-          }
-        />
```

Comment:
Please ensure that your `main React component (index.js)` is wrapped in Provider and assign store just like the example below.


    import store from './redux/configureStore';

    ReactDOM.render(
    <Provider store={store}>
        <App />
    </Provider>,
    document.getElementById('root')
    )

Reply:
Hi @micheaol according to these articles and documentation, it is ok using the `<Provider />` component as the outermost wrapping component: https://es.redux.js.org/docs/avanzado/uso-con-react-router.html#componentsrootjs , https://medium.com/@montezsmith/how-to-setup-redux-react-router-571adfde4f23#:~:text=//%20In%20../-,index.js,-import%20%7B%20Provider%20%7D%20from

Reply:
@helio3197 ,

Thank you for sharing the docs. but the project requirement expects you to follow the example below:

![image](https://user-images.githubusercontent.com/10332499/160734014-9095f0d4-9a79-4f35-93a5-9779eadf8718.png)

Kindly stick to the project requirement.

Reply:
Thanks @micheaol, but wrapping `<Provider />` just before my `<App />` component throw this error `index.tsx:24 Uncaught Error: [Provider] is not a <Route> component. All component children of <Routes> must be a <Route> or <React.Fragment>`
Remember that we are using React Router and in the example, from the instructions, the Routing is not implemented.
I do believe I'm meeting the requirement because the `<Provider />` component is currently wrapping my top-level component.
Kindly reconsider your review.

Reply:
@helio3197 ,

So sorry about the error. I believe the reason you are getting this error is that you have `routes` set-up` in your `index.js` file. Try to set all routes in your `app.js` file and only pass the `aap.js` file in your `index.js` file as you see in the image above